libs/features/dynamic-navigation.js

```diff
@@ -0,0 +1,21 @@
+import { getMetadata } from '../utils/utils.js';
+
+export default function dynamicNav(url, key) {
+  const metadataContent = getMetadata('dynamic-nav');
+
+  if (metadataContent === 'entry') {
+    window.sessionStorage.setItem('gnavSource', url);
+    window.sessionStorage.setItem('dynamicNavKey', key);
+    return url;
+  }
+
+  if (metadataContent !== 'on') return url;
+
+  if (key !== window.sessionStorage.getItem('dynamicNavKey')) return url;
+
+  const source = window.sessionStorage.getItem('gnavSource');
+
+  if (!source) return url;
+
+  return source;
```

Comment:
Since lines 12 and 14 have the same return value if the check fails, we can combine them as below
`if (metadataContent !== 'on' || key !== window.sessionStorage.getItem('dynamicNavKey')) return url;
`

Additionally, the following three lines can be reduced to one line:
  `  return window.sessionStorage.getItem('gnavSource') || url;`